// @pushprotocol/restapi/src/lib/types/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoCallStatus = exports.ChatStatus = void 0;
var ChatStatus;
(function (ChatStatus) {
    ChatStatus["ACTIVE"] = "ACTIVE";
    ChatStatus["PENDING"] = "PENDING";
    ChatStatus["ENDED"] = "ENDED";
})(ChatStatus = exports.ChatStatus || (exports.ChatStatus = {}));
var VideoCallStatus;
(function (VideoCallStatus) {
    VideoCallStatus[VideoCallStatus["UNINITIALIZED"] = 0] = "UNINITIALIZED";
    VideoCallStatus[VideoCallStatus["INITIALIZED"] = 1] = "INITIALIZED";
    VideoCallStatus[VideoCallStatus["RECEIVED"] = 2] = "RECEIVED";
    VideoCallStatus[VideoCallStatus["CONNECTED"] = 3] = "CONNECTED";
    VideoCallStatus[VideoCallStatus["DISCONNECTED"] = 4] = "DISCONNECTED";
    VideoCallStatus[VideoCallStatus["RETRY_INITIALIZED"] = 5] = "RETRY_INITIALIZED";
    VideoCallStatus[VideoCallStatus["RETRY_RECEIVED"] = 6] = "RETRY_RECEIVED";
})(VideoCallStatus = exports.VideoCallStatus || (exports.VideoCallStatus = {}));
