// @pushprotocol/restapi/src/lib/space/addSpeaker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addSpeaker = void 0;
const tslib_1 = require("tslib");
const helpers_1 = require("../chat/helpers");
const addAdmins_1 = require("../chat/addAdmins");
function addSpeaker(options) {
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        const { address } = options;
        try {
            const group = yield (0, addAdmins_1.addAdmins)({
                chatId: this.spaceSpecificData.spaceId,
                admins: [address],
                signer: this.signer,
                env: this.env,
                pgpPrivateKey: this.pgpPrivateKey,
            });
            // update space specific data
            this.setSpaceSpecificData(() => (0, helpers_1.groupDtoToSpaceDto)(group));
        }
        catch (err) {
            console.error(`[Push SDK] - API  - Error - API ${addSpeaker.name} -:  `, err);
            throw Error(`[Push SDK] - API  - Error - API ${addSpeaker.name} -: ${err}`);
        }
    });
}
exports.addSpeaker = addSpeaker;
