// @pushprotocol/restapi/src/lib/space/removeSpeaker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeSpeaker = void 0;
const tslib_1 = require("tslib");
const helpers_1 = require("../chat/helpers");
const removeAdmins_1 = require("../chat/removeAdmins");
function removeSpeaker(options) {
    return tslib_1.__awaiter(this, void 0, void 0, function* () {
        const { address } = options;
        try {
            const group = yield (0, removeAdmins_1.removeAdmins)({
                chatId: this.spaceSpecificData.spaceId,
                admins: [address],
                signer: this.signer,
                env: this.env,
                pgpPrivateKey: this.pgpPrivateKey,
            });
            // update space specific data
            this.setSpaceSpecificData(() => (0, helpers_1.groupDtoToSpaceDto)(group));
        }
        catch (err) {
            console.error(`[Push SDK] - API  - Error - API ${removeSpeaker.name} -:  `, err);
            throw Error(`[Push SDK] - API  - Error - API ${removeSpeaker.name} -: ${err}`);
        }
    });
}
exports.removeSpeaker = removeSpeaker;
